src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs

```diff
@@ -20,12 +21,16 @@ public class EnumConverter : TypeConverter
         /// Initializes a new instance of the <see cref='System.ComponentModel.EnumConverter'/> class for the given
         /// type.
         /// </summary>
-        public EnumConverter([DynamicallyAccessedMembers(TypeDescriptor.ReflectTypesDynamicallyAccessedMembers)] Type type)
+        public EnumConverter(Type type)
         {
+            if (!type.IsEnum && !type.Equals(typeof(Enum)))
```

Comment:
I'm a little bit concerned we have tests where this is used with a non-enum and it apparently works (System.Enum is really no different from just System.Object, or whatever).